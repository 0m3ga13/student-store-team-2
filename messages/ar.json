{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك"
    }
}
