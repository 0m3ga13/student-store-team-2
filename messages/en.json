{
    "Index": {
        "Title": "MiniStore",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Free delivery": "Free delivery",
        "feature descriprion 1": "You can receive your order without incurring any additional shipping costs",
        "qualite guarantee": "qualite guarantee",
        "feature descriprion 2": "Our products or services meet high standards and perform as promised.",
        "daily offers": "daily offers",
        "feature descriprion 3": "Exclusive discounts and promotions on a daily basis",
        "100% secure payement": "100% secure payement",
        "feature descriprion 4": "Your financial information is protected and your transactions are safe from unauthorized access",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "my stuff":"my stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "electronics": "electronics",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product."
    }
}
