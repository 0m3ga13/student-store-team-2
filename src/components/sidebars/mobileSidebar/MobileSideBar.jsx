"use client";
import React, { useEffect, useState } from "react";
import { AiOutlineClose } from "react-icons/ai";
import { IoOptionsSharp } from "react-icons/io5";

import ProductsFilterWrapper from "@/components/wrappers/productsFilterWrapper/ProductsFilterWrapper";

function MobileSideBar({ categories }) {
    const [isSidebarOpen, setIsSidebarOpen] = useState(false);

    const [isVisible, setIsVisible] = useState(false);

    useEffect(() => {
        // hide scrollbar when the modal is open
        if (isSidebarOpen) {
            document.body.style.overflow = "hidden";
        }

        // Function to handle the scroll event
        const handleScroll = () => {
            if (window.scrollY > 200) {
                // Show the button when scrolled down 100 pixels
                setIsVisible(true);
            } else {
                setIsVisible(false);
            }
        };

        // Attach the scroll event listener
        window.addEventListener("scroll", handleScroll);

        // Clean up the event listener when the component unmounts
        return () => {
            window.removeEventListener("scroll", handleScroll);

            // reset body overflow when the modal is close
            document.body.style.overflow = "unset";
        };
    }, [isSidebarOpen]);

    const toggleSidebar = () => setIsSidebarOpen(!isSidebarOpen);
    return (
        <div className='lg:hidden'>
            <div
                onClick={toggleSidebar}
                className={`modal fixed z-20 lg:hidden bg-[rgba(0,0,0,0.33)] z-10  top-0 left-0 w-screen h-screen ${
                    isSidebarOpen ? "scale-100" : "scale-0"
                }`}
            >
                <button className='bg-gray-100 rounded-md float-right mr-4 mt-4'>
                    <AiOutlineClose className='w-7 h-7 text-gray-400' />
                </button>
            </div>
            <div
                className={`p-10 bg-white z-20 fixed overflow-auto -translate-x-full h-screen w-3/4 top-0 left-0 transition-all ease-in  lg:translate-x-0 ${
                    isSidebarOpen ? "translate-x-0" : ""
                }`}
            >
                <ProductsFilterWrapper
                    autoSendRequest={false}
                    toggleSidebar={toggleSidebar}
                    categories={categories}
                />
            </div>
            <button
                onClick={toggleSidebar}
                className={`bg-accent w-16 h-16 text-white fixed right-3 top-[80%] transition-all ease-in flex flex-col items-center justify-center rounded-full ${
                    isVisible && !isSidebarOpen ? "scale-100" : "scale-0"
                }`}
            >
                <IoOptionsSharp className='w-7 h-7' />
                <span> Filter</span>
            </button>
        </div>
    );
}

export default MobileSideBar;
