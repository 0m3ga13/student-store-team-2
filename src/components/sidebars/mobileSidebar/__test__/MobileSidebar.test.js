import { NextIntlClientProvider } from "next-intl";
import renderer from "react-test-renderer";

import messages from "@/../messages/en.json";

import MobileSideBar from "../MobileSideBar";

// Mock the "next/navigation" module
jest.mock("next/navigation", () => ({
    useRouter: jest.fn(() => ({
        push: jest.fn(),
    })),
    useSearchParams: jest.fn(() => ({
        get: jest.fn(),
    })),
    usePathname: jest.fn(),
    useParams: () => ({
        locale: "en",
    }),
}));
it("renders correctly", () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <MobileSideBar />
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
