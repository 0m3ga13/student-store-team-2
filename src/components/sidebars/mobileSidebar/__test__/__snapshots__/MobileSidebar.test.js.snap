// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="lg:hidden"
>
  <div
    className="modal fixed z-20 lg:hidden bg-[rgba(0,0,0,0.33)] z-10  top-0 left-0 w-screen h-screen scale-0"
    onClick={[Function]}
  >
    <button
      className="bg-gray-100 rounded-md float-right mr-4 mt-4"
    >
      <svg
        className="w-7 h-7 text-gray-400"
        fill="currentColor"
        fillRule="evenodd"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M799.855 166.312c.023.007.043.018.084.059l57.69 57.69c.041.041.052.06.059.084a.118.118 0 0 1 0 .069c-.007.023-.018.042-.059.083L569.926 512l287.703 287.703c.041.04.052.06.059.083a.118.118 0 0 1 0 .07c-.007.022-.018.042-.059.083l-57.69 57.69c-.041.041-.06.052-.084.059a.118.118 0 0 1-.069 0c-.023-.007-.042-.018-.083-.059L512 569.926 224.297 857.629c-.04.041-.06.052-.083.059a.118.118 0 0 1-.07 0c-.022-.007-.042-.018-.083-.059l-57.69-57.69c-.041-.041-.052-.06-.059-.084a.118.118 0 0 1 0-.069c.007-.023.018-.042.059-.083L454.073 512 166.371 224.297c-.041-.04-.052-.06-.059-.083a.118.118 0 0 1 0-.07c.007-.022.018-.042.059-.083l57.69-57.69c.041-.041.06-.052.084-.059a.118.118 0 0 1 .069 0c.023.007.042.018.083.059L512 454.073l287.703-287.702c.04-.041.06-.052.083-.059a.118.118 0 0 1 .07 0Z"
        />
      </svg>
    </button>
  </div>
  <div
    className="p-10  bg-white z-20 fixed overflow-auto -translate-x-full h-screen w-3/4 top-0 left-0 transition-all ease-in  lg:translate-x-0 "
  >
    <div>
      <div
        className=" mb-6"
      >
        <button
          className="block text-md py-1 px-3 ml-auto mb-4 w-fit bg-gray-200 rounded-sm hover:bg-gray-400 active:bg-gray-200"
          onClick={[Function]}
        >
          Reset
        </button>
        <h1
          className="block uppercase text-lg "
        >
          Filter By
          : 
        </h1>
      </div>
      <div
        className="flex flex-col gap-8"
      >
        <div
          className="pb-10"
        >
          <h2
            className="font-jost sm:text-lg"
          >
            Price Filter
          </h2>
          <div
            className="mt-4"
          >
            <input
              className="thumb thumb--left pointer-events-none absolute h-0 w-52 outline-none z-30"
              max={1000}
              min={0}
              onChange={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "zIndex": false,
                }
              }
              type="range"
              value={0}
            />
            <input
              className="thumb thumb--right pointer-events-none absolute h-0 w-52 outline-none z-40 "
              max={1000}
              min={0}
              onChange={[Function]}
              onMouseUp={[Function]}
              type="range"
              value={1000}
            />
            <div
              className="relative w-52"
            >
              <div
                className="slider__track absolute rounded h-1 bg-gray-200 w-full z-10"
              />
              <div
                className="slider__range absolute rounded h-1 z-20 bg-[#821F40]"
              />
              <div
                className="slider__left-value absolute text-[#55585B] text-xs left-2 mt-5"
              >
                0
              </div>
              <div
                className="slider__right-value absolute text-[#55585B] text-xs mt-5 right-1"
              >
                1000
              </div>
            </div>
          </div>
        </div>
        <div
          className=""
        >
          <h2
            className="capitalize text-lg"
          >
            Product Status
          </h2>
          <div
            className="mt-4 border-t border-t-[#EEEEEE] flex flex-col gap-5 py-4 px-3"
          >
            <label
              className="group relative text-[#55585B] text-sm "
            >
              To Buy
              <input
                className="absolute opacity-0 cursor-pointer peer"
                name="radio"
                onChange={[Function]}
                type="radio"
                value="Index.sale"
              />
              <span
                className="absolute -top-2 -left-3 rounded-sm h-8 w-28 group-hover:bg-gray-200 -z-10   "
              />
            </label>
            <label
              className="group relative text-[#55585B] text-sm "
            >
              To Borrow
              <input
                className="absolute opacity-0 cursor-pointer peer"
                name="radio"
                onChange={[Function]}
                type="radio"
                value="Index.borrow"
              />
              <span
                className="absolute -top-2 -left-3 rounded-sm h-8 w-28 group-hover:bg-gray-200 -z-10   "
              />
            </label>
          </div>
        </div>
        <div>
          <h2
            className="capitalize text-lg"
          >
            Product Condition
          </h2>
          <div
            className="flex gap-2 mt-4 border-t border-t-[#EEEEEE] py-4"
          >
            <button
              className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
              onClick={[Function]}
              value={5}
            >
              5
            </button>
            <button
              className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
              onClick={[Function]}
              value={6}
            >
              6
            </button>
            <button
              className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
              onClick={[Function]}
              value={7}
            >
              7
            </button>
            <button
              className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
              onClick={[Function]}
              value={8}
            >
              8
            </button>
            <button
              className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
              onClick={[Function]}
              value={9}
            >
              9
            </button>
            <button
              className=" rounded-xl w-8 h-9   bg-white border border-gray-200"
              onClick={[Function]}
              value={10}
            >
              10
            </button>
          </div>
        </div>
        <div
          className=""
        >
          <h2
            className="capitalize text-lg"
          >
            Categories
          </h2>
          <div
            className="mt-4 border-t border-t-[#EEEEEE] flex flex-col items-start gap-5 py-4 px-3"
          />
        </div>
      </div>
      <button
        className="bg-accent  mt-4 float-right bottom-2 px-3 py-1 text-white active:bg-slate-600"
        onClick={[Function]}
      >
        Filter
      </button>
    </div>
  </div>
  <button
    className="bg-accent w-16 h-16 text-white fixed right-3 top-[80%] transition-all ease-in flex flex-col items-center justify-center rounded-full scale-0"
    onClick={[Function]}
  >
    <svg
      className="w-7 h-7"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M381.25 112a48 48 0 00-90.5 0H48v32h242.75a48 48 0 0090.5 0H464v-32zM176 208a48.09 48.09 0 00-45.25 32H48v32h82.75a48 48 0 0090.5 0H464v-32H221.25A48.09 48.09 0 00176 208zm160 128a48.09 48.09 0 00-45.25 32H48v32h242.75a48 48 0 0090.5 0H464v-32h-82.75A48.09 48.09 0 00336 336z"
      />
    </svg>
    <span>
       Filter
    </span>
  </button>
</div>
`;
